### DISCUSSION TITLE
运行Chapter3 End2End示例代码报错

### DISCUSSION THREAD

**Discussion by ykzhang-eeis** [mlc-ai/mlc-zh#176] [Q&A]
我尝试了Windows本地、WSL2子系统以及Colab，当运行到
@R.function
    def main(x: Tensor((1, 784), "float32"),
             w0: Tensor((128, 784), "float32"),
             b0: Tensor((128,), "float32"),
             w1: Tensor((10, 128), "float32"),
             b1: Tensor((10,), "float32")):
        with R.dataflow():
            lv0 = R.call_tir(linear0, (x, w0, b0), (1, 128), dtype="float32")
            lv1 = R.call_tir(relu0, (lv0,), (1, 128), dtype="float32")
            out = R.call_tir(linear1, (lv1, w1, b1), (1, 10), dtype="float32")
            R.output(out)
        return out
这段代码时，会报错error: Undefined variable: Tensor，并且会出现Dead kernel的现象，求解决方案，多谢。
![image](https://user-images.githubusercontent.com/54619352/218996553-3830b9bf-0a9c-4887-ae72-44a756486ed8.png)
![image](https://user-images.githubusercontent.com/54619352/218996648-5e107557-310f-4c2d-8257-029f4cb39159.png)

**Comment by yofufufufu** [comment]
使用`x: R.Tensor`
